experimental/python/databricks/bundles/core/_resource_mutator.py

```diff
@@ -0,0 +1,61 @@
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Generic, Type, TypeVar, overload
+
+from databricks.bundles.core._bundle import Bundle
+from databricks.bundles.core._resource import Resource
+
+if TYPE_CHECKING:
+    from databricks.bundles.jobs._models.job import Job
+
+_T = TypeVar("_T", bound=Resource)
+
+
+@dataclass(frozen=True)
+class ResourceMutator(Generic[_T]):
+    """
+    Mutators defined within a single Python module are applied in the order they are defined.
+    The relative order of mutators defined in different modules is not guaranteed.
```

Comment:
Can we enforce this somehow?

The order of `apply_mutators` in the configuration is the only order we can enforce, besides of course the natural ordering in those functions.

Reply:
The comment is outdated. It used to be true when we loaded mutators in a similar way as resources. Right now, we use order in databricks.yml, both for loading and mutating, which is more intuitive.

Reply:
Fixed here: https://github.com/databricks/cli/pull/2485